pkg/didmethod/peer/resolver.go

```diff
@@ -21,20 +22,20 @@ func NewDIDResolver(store *DIDStore) *DIDResolver {
 }
 
 // Read implements didresolver.DidMethod.Read interface (https://w3c-ccg.github.io/did-resolution/#resolving-input)
-func (resl *DIDResolver) Read(did string, versionID interface{}, versionTime string, noCache bool) ([]byte, error) {
+func (resl *DIDResolver) Read(did string, opts ...didresolver.Options) ([]byte, bool, error) {
 	// get the document from the store
 	doc, err := resl.store.Get(did)
 	if err != nil {
-		return nil, errors.Errorf("Fetching data from store failed: %w", err)
+		return nil, false, errors.Errorf("Fetching data from store failed: %w", err)
 	}
 
 	// convert the doc to JSON as DID Resolver expects byte result.
 	jsonDoc, err := doc.JSONBytes()
 	if err != nil {
-		return nil, errors.Errorf("Json marshalling of document failed: %w", err)
+		return nil, false, errors.Errorf("Json marshalling of document failed: %w", err)
 	}
 
-	return jsonDoc, nil
+	return jsonDoc, true, nil
```

Comment:
is a boolean really necessary? if jsonDoc is nil, then it means doc is not found. 
Also the boolean is set to false in case of errors only ( where jsonDoc returned is nil).
This boolean is redundant.

Reply:
I tend to agree. @troyronda thoughts?

Reply:
I think (nil, nil) is not a valid thing to return. If I get a non-nil err, I expect a valid value.

Reply:
(nil, nil) should never be returned
(!nil, !nil) should never be returned either,

the return should be either error != nil or jsonDoc != nil, the above cases should be treated as a bug

Reply:
So @Baha-sk Are you are saying that there is a NotFound error? How does the caller easily tell the error is NotFound?

Reply:
I'm not sure I'm following here.

(nil,nil) --> not found, no error
(!nil, nil) --> found, no error
(nil, !nil) --> not found, error

(!nil, !nil) should be illegal

Reply:
I dislike (nil, nil) as a signal for NotFound. I continue to think that a nil error means the value should be safe (and should exist). 

It would be helpful if someone has already documented the correct idiom to use in Go.

Reply:
from spec:
https://w3c-ccg.github.io/did-resolution/#resolving-algorithm

If the input DID does not exist, return a null result.

Reply:
@fqutishat the spec should not be used as a substitute for correct idioms in the implementation language.